testing_and_setup/compass/ocean/global_ocean/scripts/create_E3SM_coupling_files.py

```diff
@@ -307,195 +334,158 @@ def mapping_analysis(config, mesh_name, date_string, ice_shelf_cavities):
+                          ['patch', 'patc']]:
+
+        # Ocean to atmosphere
+        mapping_file = 'map_{}{}_TO_{}_{}.{}.nc'.format(
+            mesh_name, nomaskStr, atm_scrip_tag, short, date_string)
+        args = ['mpirun', '-n', nprocs, 'ESMF_RegridWeightGen',
+                '--method', method,
+                '--source', ocn_scrip_file,
+                '--destination', atm_scrip_file,
+                '--weight', mapping_file,
+                '--ignore_unmapped']
+        run_command(args)
 
-    # execute commands
-    try:
-        for method, short in [['conserve', 'aave'], ['bilinear', 'blin'], ['patch', 'patc']]:
+        # Atmosphere to ocean
+        mapping_file = 'map_{}_TO_{}{}_{}.{}.nc'.format(
+            atm_scrip_tag, mesh_name, nomaskStr, short, date_string)
+        args = ['mpirun', '-n', nprocs, 'ESMF_RegridWeightGen',
+                '--method', method,
+                '--source', atm_scrip_file,
+                '--destination', ocn_scrip_file,
+                '--weight', mapping_file,
+                '--ignore_unmapped']
+        run_command(args)
+
+    if ice_shelf_cavities:
+        print("\n Mapping files with masks for ice shelf cavities")
+        # make links
+        ocn_scrip_file = 'ocean.{}.mask.scrip.{}.nc'.format(mesh_name,
+                                                            date_string)
+        make_link('../scrip/{}'.format(ocn_scrip_file), ocn_scrip_file)
+
+        for method, short in [['conserve', 'aave'], ['bilinear', 'blin'],
+                              ['patch', 'patc']]:
 
             # Ocean to atmosphere
+            mapping_file = 'map_{}.mask_TO_{}_{}.{}.nc'.format(
+                mesh_name, atm_scrip_tag, short, date_string)
             args = ['mpirun', '-n', nprocs, 'ESMF_RegridWeightGen',
                     '--method', method,
                     '--source', ocn_scrip_file,
                     '--destination', atm_scrip_file,
-                    '--weight', 'map_' + mesh_name + nomaskStr + '_TO_' + atm_scrip_tag + '_'
-                       + short + '.' + date_string + '.nc',
+                    '--weight', mapping_file,
                     '--ignore_unmapped']
             run_command(args)
 
             # Atmosphere to ocean
+            mapping_file = 'map_{}_TO_{}.mask_{}.{}.nc'.format(
+                atm_scrip_tag, mesh_name, short, date_string)
             args = ['mpirun', '-n', nprocs, 'ESMF_RegridWeightGen',
                     '--method', method,
                     '--source', atm_scrip_file,
                     '--destination', ocn_scrip_file,
-                    '--weight', 'map_' + atm_scrip_tag + '_TO_' + mesh_name + nomaskStr + '_'
-                       + short + '.' + date_string + '.nc',
+                    '--weight', mapping_file,
                     '--ignore_unmapped']
             run_command(args)
-
-    except OSError:
-        print('mapping must be run on one compute node')
```

Comment:
This `try` / `except` around a do loop is something I would advise against.  I appreciate that there was a meaningful exception name `OSError` here.  But it would be fine to just let the `OSError` happen I think.  I could imagine various sources of `OSErrors` here, only some of which would relate to not being on a compute node.  It also is hard to figure out what went wrong if you catch the error and don't print the error message that was produced.  Which call failed?